processor/transform: empty statement list item panics during config unmarshaling

## Problem

The collector can panic while unmarshaling the transform processor config when one of the transform statement lists contains an empty YAML list item.

This was reported with `metric_statements` on `v0.153.0` and I reproduced the same nil-item path on current `main` (`350d8c128a`).

## Reproducer

The minimal shape is:

```yaml
processors:
  transform:
    metric_statements:
      -
```

In a complete config:

```yaml
receivers:
  nop:

processors:
  transform:
    metric_statements:
      -

exporters:
  nop:

service:
  pipelines:
    metrics:
      receivers: [nop]
      processors: [transform]
      exporters: [nop]
```

An empty item decodes to `nil`, and `Config.Unmarshal` currently calls `reflect.TypeOf(value).Kind()` for each item without checking for nil:

```go
for _, value := range values {
    if reflect.TypeOf(value).Kind() == reflect.String {
        basicStatements = append(basicStatements, value)
    } else {
        ...
    }
}
```

## Actual Behavior

The collector panics during config unmarshaling:

```text
panic: runtime error: invalid memory address or nil pointer dereference

github.com/open-telemetry/opentelemetry-collector-contrib/processor/transformprocessor.(*Config).Unmarshal(...)
    github.com/open-telemetry/opentelemetry-collector-contrib/processor/transformprocessor@v0.153.0/config.go:103
```

## Expected Behavior

The processor should return a configuration validation/unmarshal error instead of panicking. For example, an empty item in `metric_statements` should be reported as an invalid statement entry.

## Scope

This does not appear to be specific to `metric_statements`. A temporary test that unmarshaled `[]any{nil}` for each statement field showed the same panic for:

- `trace_statements`
- `metric_statements`
- `log_statements`
- `profile_statements`

The fix likely belongs in the shared `Config.Unmarshal` loop for all four fields.

--- a/.chloggen/49245-transformprocessor-empty-statement.yaml
+++ b/.chloggen/49245-transformprocessor-empty-statement.yaml
@@ -1,0 +1,5 @@
+change_type: bug_fix
+component: processor/transform
+note: Fix transform processor config unmarshaling to return an error for empty statement list items instead of panicking.
+issues: [49245]
+change_logs: [user]

--- a/processor/transformprocessor/config.go
+++ b/processor/transformprocessor/config.go
@@ -99,6 +99,9 @@ func (c *Config) Unmarshal(conf *confmap.Conf) error {
 		statementsConfigs := make([]any, 0, len(values))
 		var basicStatements []any
 		for _, value := range values {
+			if value == nil {
+				return fmt.Errorf("invalid %s item: empty statement list items are not supported", fieldName)
+			}
 			// Array of strings means it's a basic configuration style
 			if reflect.TypeOf(value).Kind() == reflect.String {
 				basicStatements = append(basicStatements, value)

--- a/processor/transformprocessor/config_test.go
+++ b/processor/transformprocessor/config_test.go
@@ -11,6 +11,7 @@ import (
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/confmap"
 	"go.opentelemetry.io/collector/confmap/confmaptest"
 	"go.opentelemetry.io/collector/confmap/xconfmap"
 
@@ -404,4 +405,26 @@ func Test_MixedConfigurationStyles(t *testing.T) {
 	sub, err := cm.Sub(component.NewIDWithName(metadata.Type, "mixed_configuration_styles").String())
 	require.NoError(t, err)
 	assert.ErrorContains(t, sub.Unmarshal(cfg), "configuring multiple configuration styles is not supported")
+}
+
+func Test_EmptyStatementListItem(t *testing.T) {
+	t.Parallel()
+
+	for _, fieldName := range []string{
+		"trace_statements",
+		"metric_statements",
+		"log_statements",
+		"profile_statements",
+	} {
+		t.Run(fieldName, func(t *testing.T) {
+			t.Parallel()
+
+			cfg := NewFactory().CreateDefaultConfig()
+			conf := confmap.NewFromStringMap(map[string]any{
+				fieldName: []any{nil},
+			})
+
+			require.ErrorContains(t, conf.Unmarshal(cfg), "invalid "+fieldName+" item: empty statement list items are not supported")
+		})
+	}
 }

